packages/flutter_tools/test/data/dart_dependencies_test/syntax_error/foo.dart

```diff
@@ -2,4 +2,4 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-bad programmer!
+import bad programmer!
```

Comment:
The old sky_snapshot based system would detect many syntax errors beyond imports on the host and not send it to the device for parsing. I think it's fine trading faster hot reload times for more device side errors.